lib/ruby_llm/structured_output.rb

```diff
@@ -0,0 +1,205 @@
+        end
+
+        def object(name = nil, description: nil, &block)
+          sub_schema = Class.new(Schema)
+          sub_schema.class_eval(&block)
+
+          schema = {
+            type: 'object',
+            properties: sub_schema.properties,
+            required: sub_schema.required,
+            additionalProperties: false,
+            description: description
+          }.compact
+
+          name ? add_property(name, schema) : schema
+        end
+
+        def array(name, type = nil, description: nil, &block)
+          items = if block_given?
+                    collector = SchemaCollector.new
+                    collector.instance_eval(&block)
+                    collector.schemas.first
+                  elsif type.is_a?(Symbol)
+                    case type
+                    when :string, :number, :boolean, :null
+                      send(type)
+                    else
+                      ref(type)
+                    end
+                  else
+                    raise ArgumentError, "Invalid array type: #{type}"
+                  end
+
+          add_property(name, {
+            type: 'array',
+            description: description,
+            items: items
+          }.compact)
+        end
+
+        def any_of(name, description: nil, &block)
+          collector = SchemaCollector.new
+          collector.instance_eval(&block)
+
+          add_property(name, {
+            description: description,
+            anyOf: collector.schemas
+          }.compact)
+        end
+
+        def ref(schema_name)
+          { '$ref' => "#/$defs/#{schema_name}" }
+        end
+
+        def properties
+          @properties ||= {}
+        end
+
+        def required
+          @required ||= []
+        end
+
+        def definitions
+          @definitions ||= {}
+        end
+
+        def define(name, &)
+          sub_schema = Class.new(Schema)
+          sub_schema.class_eval(&)
+
+          definitions[name] = {
+            type: 'object',
+            properties: sub_schema.properties,
+            required: sub_schema.required
+          }
+        end
+
+        private
+
+        def add_property(name, definition)
+          properties[name.to_sym] = definition
+          required << name.to_sym
+        end
+      end
+
+      # Simple collector that just stores schemas
+      class SchemaCollector
+        attr_reader :schemas
+
+        def initialize
+          @schemas = []
+        end
+
+        def method_missing(method_name, ...)
+          if Schema.respond_to?(method_name)
+            @schemas << Schema.send(method_name, ...)
+          else
+            super
+          end
+        end
+
+        def respond_to_missing?(method_name, include_private = false)
+          Schema.respond_to?(method_name) || super
+        end
+      end
+
+      def initialize(name = nil)
+        @name = name || self.class.name
+        validate_schema
+      end
+
+      def to_hash
```

Comment:
One change and it will be compatible with my PR https://github.com/crmne/ruby_llm/pull/122

```suggestion
      def json_schema
```